netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,243 @@
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          val scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+        })
+      }
+    } catch {
+      case e: Exception => throw new Exception(s"Error in showJsonInDialog: ${e.getMessage}", e)
+    }
+  }
+
+
+  // Lazily initialize the Markdown parser and renderer (can be reused for multiple calls)
+  private lazy val markdownParser = Parser.builder().build()
+  private lazy val htmlRenderer = HtmlRenderer.builder().build()
+
+  def formatJsonObjectList(jsonObjectList: List[JsonObject]): String = {
+    jsonObjectList.map { obj =>
+      // Convert fullText from Markdown to HTML
+      val fullTextHtml = htmlRenderer.render(markdownParser.parse(obj.fullText))
+
+      // Format title, date, and the converted HTML fullText
+      s"<h3>* ${I18N.gui.get("dialog.interface.update")}: ${obj.date} -- ${obj.title}</h3>" +
+        s"$fullTextHtml"
+    }.mkString("<html>", "", "</html>")
+  }
+
+  def getJsonObjectHead: Option[String] = {
+    // Check if jsonObjectList is null or empty, and fetch the JSON if needed
+    if (jsonObjectList == null) {
+      val jsonContent = fetchJsonFromUrl(JsonUrl) // Use the static URL here
+      jsonObjectList = parseJsonToList(jsonContent)
+    }
+
+    // Return the title of the first item if jsonObjectList is not null and has elements
+    jsonObjectList match {
+      case null | Nil =>
+        println("jsonObjectList is empty or Nil; hiding NotificationBanner.")
+        this.setVisible(false) // Hide the NotificationBanner panel
+        None // Return None if jsonObjectList is null or empty
+      case list =>
+        this.setVisible(true) // Ensure the NotificationBanner is visible
+        list.headOption.map(_.title) // Return the title of the head element if available
+    }
+  }
+
+
+  private def isShowNeeded(): Boolean = {
+    val prefs = JPreferences.userNodeForPackage(getClass)
+    val jsonContent = fetchJsonFromUrl(JsonUrl)
+    jsonObjectList = parseJsonToList(jsonContent) // Populate the class variable
+    val lastSeenEventId = prefs.getInt(lastSeenEventIdKey, -1); // Returns -1 if "event-id" is not found
+    if(jsonObjectList.isEmpty ) {
```

Comment:
These next two if statements are pretty verbose, you could combine them into one check that is directly returned and it will be easier to read and interpret.

Reply:
Good call; I've reduced it to a one-liner:
jsonObjectList.nonEmpty && jsonObjectList.head.eventId > lastSeenEventId